### PR TITLE
Consts and for-loops (under EcmaScript5.1 profile)

### PR DESCRIPTION
**Pull request by vsimko**
- cleaned some for-loops
- converted some vars to consts
(all under EcmaScript5.1 profile in IntelliJ IDEA)
related to #178

### CODE REVIEW

**Comment by mvertes** [comment]
Closing this one, as I applied pull request regarding use of '==='. The for-loop issue to be discussed separately.
Thanks

File: lib/client.js [hunk 1]
```diff
@@ -256,15 +270,15 @@ function Client(opt, callback) {
...
       self.id = data.id;
       self.uuid = data.uuid;
       self.emit('connect', data);
-      if (data.devices) {   // Cache remote ids
-        for (i = 0; i < data.devices.length; i++) {
-          d = data.devices[i];
+
+      // Cache remote ids
+      if (data.devices) {
+        for (const d in data.devices) {
```
Comment (mvertes) [inline lib/client.js]: Is it correct ? Maybe you confuse with array.forEach.
Arrays should be almost always iterated with a numerical index. See this [jsperf](https://jsperf.com/fast-array-foreach)

To restrict the scope of local iterator, we should use ECMA 5.1 `let` statement. It should be something like:

```javascript
for (let i = 0; i < data.devices.length; i++) {
  let d = data.devices[i];
  ...
}
````

File: lib/client.js [hunk 2]
```diff
@@ -334,10 +348,11 @@ Client.prototype.send = thenify(function (uuid, msg, callback) {
 });
 
 Client.prototype.devices = thenify(function (o, callback) {
-  var self = this;
+  const self = this;
   this.send(0, {cmd: 'devices', data: {query: o}}, function (err, dev) {
-    for (var i in dev)
-      self.hostId[dev[i].uuid] = dev[i].id;
+    for (const d in dev) {
```
Comment (mvertes) [inline lib/client.js]: incorrect, see above comment about for loop, and our discussion in #178

File: lib/client.js [hunk 3]
```diff
@@ -410,12 +426,18 @@ Client.prototype.createStreamTo = function (msg) {
...
 Client.prototype.createStreamFrom = function (client, msg) {
-  var sid, s, self = this;
-  for (sid = 0; sid < streamMaxId; sid++)
+  var sid;
+  for (sid = 0; sid < streamMaxId; sid++) {
     if (this.consumer.streams[sid] === undefined) break;
-  if (sid == streamMaxId) throw 'no more free streamid';
-  this.consumer.streams[sid] = s = new stream.PassThrough();
-  s.once('end', function () {self.consumer.streams[sid] = undefined;});
+  }
```
Comment (mvertes) [inline lib/client.js]: may be ok. Should be proposed in a separate pull request

File: lib/context.js [hunk 4]
```diff
@@ -47,16 +49,19 @@ function SkaleContext(arg) {
...
+
     process.title = 'skale-master_' + data.devices[0].wsid + ' ' + __filename;
-    if (!nworker || nworker > data.devices.length)
+    if (!nworker || nworker > data.devices.length) {
       nworker = data.devices.length;
+    }
+
     log('workers:', nworker);
-    for (i = 0; i < nworker; i++) {
+    for (var i = 0; i < nworker; i++) {
```
Comment (mvertes) [inline lib/context.js]: use `let i = 0; ...` instead of `var i = 0; ...`, to set the scope of `i` to the loop bloc

File: lib/context.js [hunk 5]
```diff
@@ -155,53 +160,58 @@ function SkaleContext(arg) {
       var nodeId;
       var p = task.pid;
       var node = task.nodes[task.datasetId];
-      var part = node.shufflePartitions ? node.shufflePartitions[p] : node.partitions[p];
-      var pindex = {};
+      var part = node.shufflePartitions
```
Comment (mvertes) [inline lib/context.js]: Ok with the multi-line ternary operator.

**Comment by vsimko** [comment]
Yeah, I was (again) a bit too excited :)
I should have definitely split the changes properly, so I'll try to fix that next time.
